src/include/btmem.h

```diff
@@ -489,8 +489,20 @@ struct __wt_page_modify {
 	WT_SPINLOCK page_lock;		/* Page's spinlock */
 
 	/*
-	 * The write generation is incremented when a page is modified, a page
-	 * is clean if the write generation is 0.
+	 * The write generation is incremented when a page is modified.
+	 *
+	 * 0: The page is clean.
+	 * 1: The page has 1 unreconciled update.
+	 * 2: The page has 2 or more unreconciled updates.
+	 *
+	 * When we reconcile, we artificially set the write generation to 1 and
+	 * check afterwards that the write generation hasn't changed (incr to 2)
+	 * to determine whether reconciliation was able to render the page
+	 * clean.
+	 *
+	 * Using write generation as a simple counter of the number of updates
+	 * is not acceptable as some workloads will get close to hitting
+	 * UINT32_MAX.
```

Comment:
If this is the direction that this change is going in, then I'd like to see it move that way all the way. That is, `write_gen` is really not a generation number anymore. It is a state now and should be coded as such. So the field could be named `mod_state` or `page_state` or something similar and the hard-coded numbers should be defined as `..._STATE_CLEAN`, `..._STATE_SINGLE` and `..._STATE_MANY`. I think it will be a simpler concept and the comments will read clearer once it has a slant of state instead of write generation.

Reply:
8a50aa7